src/excmds.ts: Implement :sanitize.

glacambre opened the pull request:
This implements part of Vimperator/Pentadactyl's :sanitize function.

Three things that should be discussed: Some items can be deleted under Chrome but not under Firefox. Should support for deleting these items be added to Tridactyl? It really wouldn't complicate the code much (just a three line long `if` condition).

Deleting the "pluginData" item makes a lot of errors pop up in the console. Should we prevent users from deleting this item at all? It might be useful if the user really needs to reset his Tridactyl profile, but it can be very dangerous too since there's no way to really backup your settings yet. Maybe we should let the user delete it and ask for confirmation?

I also didn't implement the timespan feature vimperator and pentadactyl have. I'm unsure about how to proceed for command line flags. Let me know how you want it to be implemented and how it should look.

This will fix https://github.com/cmcaine/tridactyl/issues/132 once it is merged.

bovine3dom commented:
I'd like see more sanitisation for Tridactyl related things.

The easiest way to do it would be to do `browser.storage.{local,sync}.clear()`.

I don't think we need to warn people particularly strongly about losing their settings - I think it's probably expected behaviour.

glacambre commented:
I added the requested function calls but I'm unsure about to how to test them, please let me know what I could try.

Also please let me know before you plan on merging this PR, I'll squash the commits, the history is kinda ugly right now.

glacambre commented:
I incorporated the changes you suggested, added a way to specify a timespan and added comments. This PR is ready to be merged in my opinion, please let me know what you think.
